x/valset/keeper/keeper.go

```diff
@@ -525,14 +549,21 @@ func (k Keeper) GetSigningKey(ctx sdk.Context, valAddr sdk.ValAddress, chainType
 }
 
 // IsJailed returns if the current validator is jailed or not.
-func (k Keeper) IsJailed(ctx sdk.Context, val sdk.ValAddress) bool {
-	return k.staking.Validator(ctx, val).IsJailed()
+func (k Keeper) IsJailed(ctx context.Context, val sdk.ValAddress) (bool, error) {
+	a, err := k.staking.Validator(ctx, val)
+	if err != nil {
+		return a.IsJailed(), err
+	}
+	return a.IsJailed(), nil
 }
 
-func (k Keeper) Jail(ctx sdk.Context, valAddr sdk.ValAddress, reason string) error {
+func (k Keeper) Jail(_ctx context.Context, valAddr sdk.ValAddress, reason string) error {
+	ctx := sdk.UnwrapSDKContext(_ctx)
 	jailTime := ctx.BlockTime()
-	val := k.staking.Validator(ctx, valAddr)
-
+	val, err := k.staking.Validator(ctx, valAddr)
+	if err != nil {
+		return ErrValidatorWithAddrNotFound.Format(valAddr)
```

Comment:
Why would you not return the actual error?

Reply:
Actual `err` and `ErrValidatorWithAddrNotFound` specify the same thing ( i.e validator does not exist).So i used the ErrValidatorWithAddrNotFound error instead of actual error

Reply:
Okay
I've returned the actual error.

Reply:
@byte-bandit Can you resolve the completed conversation